packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -484,9 +478,50 @@ class _TextSelectionHandleOverlay extends StatefulWidget {
   _TextSelectionHandleOverlayState createState() => _TextSelectionHandleOverlayState();
 }
 
-class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay> {
+class _TextSelectionHandleOverlayState
+    extends State<_TextSelectionHandleOverlay> with SingleTickerProviderStateMixin {
   Offset _dragPosition;
 
+  AnimationController _controller;
+  Animation<double> get _opacity => _controller.view;
+
+  ValueListenable<bool> get _visibility {
+    switch (widget.position) {
+      case _TextSelectionHandlePosition.start:
+        return widget.renderObject.selectionStartInViewport;
+      case _TextSelectionHandlePosition.end:
+        return widget.renderObject.selectionEndInViewport;
+    }
+    return null;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    _controller = AnimationController(duration: TextSelectionOverlay.fadeDuration, vsync: this);
+
+    if (_visibility.value) {
+      _controller.forward();
+    }
+    _visibility.addListener(_handleVisibilityChanged);
```

Comment:
you don't handle the case of `widget.position` changing.

You probably want a didUpdateWidget that unsubscribes from the old one and subscribes to the new one if it changed.

Reply:
For my own enlightenment, how can this happen? When the selection handles are built, either none are built, only the left handle, or both handles. I don't see how it can happen that the left handle switches to use state from the right handle or vice versa.

Reply:
At the end of the day _TextSelectionHandleOverlay is just another regular widget and it should conform to the widget protocol, which allows that a state object can be attached to an arbitrary new configuration (e.i. Widget). This new Widget may have a different position.

Reply:
Thanks for the explanation. Done.